funnel/models/proposal.py

```diff
@@ -273,6 +343,26 @@ def permissions(self, user, inherited=None):
                     perms.add('decline-proposal')  # Decline speaking
         return perms
 
+    # Roles
+
+    def roles_for(self, user=None, token=None):
+        if not user and not token:
+            return set()
+        roles = super(Proposal, self).roles_for(user, token)
+        if self.speaker and self.speaker == user:
+            roles.add('speaker')
+        if self.user == user:
+            roles.add('proposer')
+        if self.proposal_space.admin_team in user.teams or self.proposal_space.profile.admin_team in user.teams:
+            roles.add('admin')
+        if self.proposal_space.review_team in user.teams:
+            roles.add('reviewer')
+            roles.add('admin')
```

Comment:
Are you sure you want to add the `admin` role here? There won't be any difference between a user in the `review_team` and a user in the `admin_team` in that case.

Reply:
@shreyas-satish right. my bad. Question: Should a reviewer be able to change state of proposals? Right now they can.

Reply:
> Should a reviewer be able to change state of proposals?
Yes.